extensions/integration-tests/src/test/utils.ts

```diff
@@ -180,6 +180,27 @@ export async function runQuery(query: string, ownerUri: string): Promise<azdata.
 
 }
 
+export async function retryFunction(fn: () => Promise<any>, retryCount: number): Promise<any> {
+	let result: any;
+	while (retryCount > 0) {
```

Comment:
If it hits the retry count times without succeeding it should throw an exception